docs/architecture/adr-043-nft-module.md

```diff
@@ -35,341 +36,224 @@ The current design is based on the work done by [IRISnet team](https://github.co
-  m.keeper.AssertTypeNotExist(msg.Metadata.Type)
-
-  bz := m.keeper.cdc.MustMarshalBinaryBare(msg.Metadata)
-  typeStore := m.keeper.getTypeStore(ctx)
-  typeStore.Set(msg.Type, bz)
-  
-  bz := m.keeper.cdc.MustMarshalBinaryBare(msg.Issuer)
-  typeOwnerStore := m.keeper.getTypeOwnerStore(ctx)
-  typeOwnerStore.Set(msg.Type, bz)
-  
-  return &types.MsgIssueResponse{}, nil
-}
-
-func (m msgServer) Mint(ctx context.Context, msg *types.MsgMint) (*types.MsgMintResponse, error) {
-  m.keeper.AssertTypeExist(msg.NFT.Type)
-  m.keeper.AssertCanMint(msg.NFT.Type, msg.Minter)
-  
-  baseDenom := fmt.Sprintf("%s-%s", msg.NFT.Type, msg.NFT.Id)
-  bkMetadata := bankTypes.Metadata{
-    Symbol:      metadata.Symbol,
-    Base:        baseDenom,
-    Name:        metadata.Name,
-    URI:         msg.NFT.URI,
-    Description: metadata.Description,
-  }
-  
-  m.keeper.bank.SetDenomMetaData(ctx, bkMetadata)
-  mintedCoins := sdk.NewCoins(sdk.NewCoin(baseDenom, 1))
-  m.keeper.bank.MintCoins(types.ModuleName, mintedCoins)
-  m.keeper.bank.SendCoinsFromModuleToAccount(types.ModuleName, msg.Owner, mintedCoins)
-  
-  bz := m.keeper.cdc.MustMarshalBinaryBare(&msg.NFT)
-  
-  nftStoreByType := m.keeper.getNFTStoreByType(ctx, msg.NFT.Type)
-  nftStoreByType.Set(msg.NFT.Id, bz)
-  
-  nftStore := m.keeper.getNFTStore(ctx)
-  nftStore.Set(baseDenom, bz)
-  
-  return &types.MsgMintResponse{}, nil
-}
-
-func (m msgServer) Edit(ctx context.Context, msg *types.MsgEdit) (*types.MsgEditResponse, error) {
-  m.keeper.AssertNFTExist(msg.Type, msg.Id)
-  m.keeper.AssertCanEdit(msg.Type, msg.Id, msg.Editor)
-
-  bz := m.keeper.cdc.MustMarshalBinaryBare(&msg.NFT)
-  
-  nftStoreByType := m.keeper.getNFTStoreByType(ctx, msg.NFT.Type)
-  nftStoreByType.Set(msg.NFT.Id, bz)
-  
-  return &types.MsgEditResponse{}, nil
-}
-
 func (m msgServer) Send(ctx context.Context, msg *types.MsgSend) (*types.MsgSendResponse, error) {
-  m.keeper.AssertNFTExist(msg.Type, msg.Id)
-
-  sentCoins := sdk.NewCoins()  
-  baseDenom := fmt.Sprintf("%s-%s", nft.Type, nft.Id)
-  sentCoins = sentCoins.Add(sdk.NewCoin(baseDenom, 1)) 
-  m.keeper.bank.SendCoins(ctx, msg.Sender, msg.Reveiver, sentCoins)
-  
-  return &types.MsgSendResponse{}, nil
-}
-
-func (m Keeper) Burn(ctx sdk.Context, msg *types.MsgBurn) (types.MsgBurnResponse,error) {
-  m.keeper.AssertNFTExist(msg.Type, msg.Id)
-
-  nftStoreByType := m.keeper.getNFTStoreByType(ctx, msg.Type)
-  nft := nftStoreByType.Get(msg.Id)
+  // check current ownership
+  assertEqual(msg.Sender, m.k.GetNftOwner(msg.Genre, msg.Id))
 
-  baseDenom := fmt.Sprintf("%s-%s", msg.Type, msg.Id)
-  coins := sdk.NewCoins(sdk.NewCoin(baseDenom, 1))
-  m.keeper.bank.SendCoinsFromAccountToModule(ctx, msg.Destroyer, types.ModuleName, coins)
-  m.keeper.bank.BurnCoins(ctx, types.ModuleName, coins)
+  // change ownership mapping
+  m.k.SetNftOwner(msg.Genre, msg.Id, msg.Receiver)
```

Comment:
Might be good to make this `m.k.TransferTo(msg.Genre, msg.Id, msg.Receiver)` to distinguish from `m.k.Mint(msg.Genre, msg.Id, msg.Receiver)` which should affect `totalSupply` whereas `TransferTo` doesn't.